homeassistant/components/tahoma/lock.py

```diff
@@ -0,0 +1,87 @@
+"""Support for Tahoma lock."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    ATTR_NAME,
+    STATE_LOCKED,
+    STATE_UNLOCKED,
+)
+
+from . import DOMAIN as TAHOMA_DOMAIN, TahomaDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Tahoma lock."""
+    controller = hass.data[TAHOMA_DOMAIN]["controller"]
+    devices = []
+    for device in hass.data[TAHOMA_DOMAIN]["devices"]["lock"]:
+        devices.append(TahomaLock(device, controller))
+    add_entities(devices, True)
+
+
+class TahomaLock(TahomaDevice, LockDevice):
+    """Representation a Tahoma lock."""
+
+    def __init__(self, tahoma_device, controller):
+        """Initialize the device."""
+        super().__init__(tahoma_device, controller)
+        self._lock_status = None
+        self._available = False
+        self._battery_level = None
+        self._name = None
```

Comment:
not sure what you mean exactly (I'm usually not a python dev). Is it this kind of thing?

```
    @property
    def name(self):
        """Return the name of the lock"""
        return self._name

    @property
    def available(self):
        """Return the availability of the lock"""
        return self._available
```

Reply:
if it is, why not adding a battery level `@property` as well?

Reply:
> Is it this kind of thing?

Yes exactly.

> if it is, why not adding a battery level @property as well?

Because the parent class LockDevice does not have such an attribute and therefore it should be added as an attribute.
Furthermore, other lock platforms do the same.

Reply:
understood, thanks :)